test/CMakeLists.txt

```diff
@@ -23,6 +23,12 @@ function(linkAndDiscoverTest basename)
     gtest_discover_tests(${basename} ${basename})
 endfunction()
 
+function(linkAndDiscoverTestSerial basename)
+    linkTest(${basename} ${ARGN})
+    gtest_discover_tests(${basename} ${basename} PROPERTIES RUN_SERIAL
+            TRUE)
+endfunction()
+
```

Comment:
I dislike this doubling of the function. But it is what I could come up with my limited CMake knowledge. But I think it should be possible to this in function and still keep the `ARGN` intact with some more work.